src/groups.rs

```diff
@@ -41,16 +42,66 @@ impl Groups {
                 .path()
                 .strip_prefix(&group_dir)?
                 .to_str()
-                .ok_or(anyhow!("Will not fail on Linux"))?
+                .ok_or(anyhow!("will not fail on linux"))?
                 .to_string();
 
             log::info!("parsing group file: {group_name}@{group_file:?}");
 
-            let file_contents = read_to_string(group_file.path()).context("Reading group file")?;
+            let file_contents = read_to_string(group_file.path()).context("reading group file")?;
+
+            let install_options: InstallOptions =
+                parse_group_file(&group_name, &file_contents).context("parsing group file")?;
 
-            let packages: PackagesInstall = toml::from_str(&file_contents)?;
-            groups.insert(group_name, packages);
+            groups.insert(group_name, install_options);
         }
         Ok(groups)
     }
 }
+
+fn parse_group_file(group_name: &str, contents: &str) -> Result<InstallOptions> {
+    let mut install_options = InstallOptions::default();
+
+    let toml = toml::from_str::<Table>(contents)?;
+
+    for (key, value) in toml.iter() {
+        install_options.append(&mut parse_toml_key_value(group_name, key, value)?);
+    }
+
+    Ok(install_options)
+}
+
+fn parse_toml_key_value(group_name: &str, key: &str, value: &Value) -> Result<InstallOptions> {
+    macro_rules! x {
+        ($($backend:ident),*) => {
+            $(
+                if key.to_lowercase() == $backend.to_string().to_lowercase() {
+                    let mut install_options = InstallOptions::default();
+
+                    let packages = value.as_array().context(
+                        anyhow!("the {} backend in the {group_name} group toml file has a non-array value", $backend)
+                    )?;
+
+                    for package in packages {
+                        match package {
+                            Value::String(x) => {
+                                install_options.$backend.insert(x.clone(), <$backend as Backend>::InstallOptions::default());
+                            },
+                            Value::Table(x) => {
+                                install_options.$backend.insert(toml::from_str(&x.to_string()).context("parsing package install options")?, <$backend as Backend>::InstallOptions::default());
```

Comment:
This doesn't work. I think what you wanted instead was to have the key value pairs inserted. For that we need to `impl From<toml::map::Map<.... for ArchInstallOptions` and other install options as well.
This fails atm,

Reply:
Something I can think of is this
```
install_options.$backend.insert(x.get("package")?.as_str()?.to_string(), 
                               <$backend as Backend>::InstallOptions::from(x));
```

Reply:
I don't think conversion via `From` is idiomatic since `From` is for guaranteed conversions whereas parsing can fail, maybe `FromStr` instead.

Reply:
I've switched to a new per-backend trait method `try_parse_toml_package()` and used an odd serde method that seems to be working with this test group file:
```toml
paru = [
	"pacdef",
	{ package = "pacdef" },
	{ package = "pacdef", optional_deps = [
		"rustup",
	] },
]

cargo = [
	"pacdef",
	{ package = "pacdef" },
	{ package = "pacdef", all_features = true },
]
```